nav2_collision_monitor/src/scan.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2022 Samsung Research Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_collision_monitor/scan.hpp"
+
+#include <cmath>
+
+namespace nav2_collision_monitor
+{
+
+Scan::Scan(
+  const nav2_util::LifecycleNode::WeakPtr & node,
+  std::shared_ptr<tf2_ros::Buffer> tf_buffer,
+  const std::string source_name,
+  const std::string base_frame_id,
+  const double transform_tolerance,
+  const double max_time_shift)
+{
+  node_ = node;
+  auto node_sptr = node_.lock();
+  if (node_sptr) {
+    RCLCPP_INFO(node_sptr->get_logger(), "Creating Scan");
+  }
+
+  tf_buffer_ = tf_buffer;
+
+  source_type_ = SCAN;
+  source_name_ = source_name;
+  base_frame_id_ = base_frame_id;
+
+  transform_tolerance_ = transform_tolerance;
+  max_time_shift_ = max_time_shift;
+}
+
+Scan::~Scan() {
+  auto node = node_.lock();
+  if (node) {
+    RCLCPP_INFO(node->get_logger(), "Destroying Scan");
```

Comment:
As well with the logger / all logs using `source_name_`

Reply:
[Done](https://github.com/ros-planning/navigation2/pull/2982/files#diff-7e24f9dbbee0e03a22353c8e2625792fc824ced725395a85665629f232ce4706R101)